xlators/features/cloudsync/src/cloudsync.c

```diff
@@ -293,6 +294,7 @@ cs_readdirp(call_frame_t *frame, xlator_t *this, fd_t *fd, size_t size,
                    "dict");
             goto err;
         }
+        xref = _gf_true;
```

Comment:
To make it consistent with all other places, we should take a ref if `xdata` already exists.

Reply:
You said "ok", but the code is not modified. Is there any reason ?